more information about mqtt performance impact considerations & update a file name (#2170)
theforeman/foreman-documentation: guides/common/modules/proc_rex-pull-based-transport-tuning.adoc

[id="REX_pull_based_transport_tuning_{context}"]

> = REX pull-based transport tuning

Reviewer: ```suggestion
= Increasing Host Limit for Pull-Based REX Transport
```
Please, also rename the ID and file name. And watch the capitalization of the title.

Author: done. thanks

> = REX pull-based transport tuning

Reviewer 2: ```suggestion
= Tuining REX Pull-Based Transport
```

Please also adjust the anchor and file name. Procedures (almost?) always start with a verb in gerund form.

Reviewer 3: With `s/Tuining/Tuning/`, I agree.

{Project} has a pull-based transport that has the features of REX but with a deployment model similar to katello-agent. 
This transport uses MQTT as its messaging protocol and includes an MQTT client running on each host.

> In order to tune the MQTT server (`mosquitto`) and increase the number of hosts connected to it, you can follow the following procedure on the {ProjectServer}:

Reviewer 4: ```suggestion
The following procedure tunes the MQTT server (`mosquitto`) and increases the number of hosts connected to it on the {ProjectServer}:
```

.Procedure

. Enable pull-based remote execution on your {ProjectServer} or {SmartProxyServer}:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> {foreman-installer} --foreman-proxy-plugin-remote-execution-script-mode pull-mqtt

Reviewer 2: This is identical to the file above. What's the reasoning? Do you want to show both increasing and decreasing?

----
+
Note that your {ProjectServer} or {SmartProxyServer} can only use one transport mode, either SSH or MQTT.
. To increase the default number of hosts accepted by the MQTT server, create the following system unit definition file and set its contents as specified in the below given example (this will allow the `mosquitto` service to handle ~5000 hosts):
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> cat >/etc/systemd/system/mosquitto.service.d/limits.conf <<__EOF__

Reviewer 5: I think this one is invalid because the installer will purge this file.

And even if it is, I think we need an issue to track removal of this workaround. https://github.com/eclipse/mosquitto/issues/2674 isn't getting any traction so if this is needed, an installer issue to manage this is needed.
